ark/phenotyping/som_utils.py

```diff
@@ -67,6 +68,136 @@ def compute_cluster_avg(fovs, channels, base_dir, cluster_col,
     return sum_count_totals
 
 
+def compute_cell_cluster_avg(cluster_path, column_prefix, cluster_col):
+    """For each cell SOM cluster, compute the average number of associated pixel SOM/meta clusters
+
+    Args:
+        cluster_path (str):
+            The path to the cell data with SOM labels
+        column_prefix (str):
+            The prefix of the columns to subset, should be 'cluster' or 'hCluster_cap'
+        cluster_col (str):
+            Name of the cell cluster column to group by
+
+    Returns:
+        pandas.DataFrame:
+            Contains the average values for each column across cell SOM clusters
+    """
+
+    # read in the clustered data
+    cluster_data = feather.read_dataframe(cluster_path)
+
+    # subset by columns with cluster in them
+    column_subset = [c for c in cluster_data.columns.values if c.startswith(column_prefix + '_')]
+    cluster_data_subset = cluster_data.loc[:, column_subset + [cluster_col]]
+
+    # average each column grouped by the cell SOM cluster column
+    mean_count_totals = cluster_data_subset.groupby(cluster_col).mean().reset_index()
+
+    return mean_count_totals
+
+
+def compute_cell_cluster_counts(fovs, channels, base_dir, consensus_dir,
+                                cell_table_path, cluster_col='cluster'):
+    """Create a matrix with each fov-cell label pair and their pixel SOM/meta cluster counts
+
+    Args:
+        fovs (list):
+            The list of fovs to subset on
+        channels (list):
+            The list of channels to subset on
```

Comment:
I might be missing something but why is channels included here?

Reply:
It should not be, it's been removed.